File: data/cdk-dynamo/src/main/java/com/josue/cdk/data/CdkDynamoStack.java
package com.josue.cdk.data;

import software.constructs.Construct;
import software.amazon.awscdk.Stack;
import software.amazon.awscdk.StackProps;
import software.amazon.awscdk.services.dynamodb.Attribute;
import software.amazon.awscdk.services.dynamodb.AttributeType;
// import software.amazon.awscdk.Duration;
// import software.amazon.awscdk.services.sqs.Queue;
import software.amazon.awscdk.services.dynamodb.Table;
import software.amazon.awscdk.services.dynamodb.TableProps;

public class CdkDynamoStack extends Stack {
    public CdkDynamoStack(final Construct scope, final String id) {
        this(scope, id, null);
    }

    public CdkDynamoStack(final Construct scope, final String id, final StackProps props) {
        super(scope, id, props);

        Table customerTable = new Table(this, "customer-table", TableProps.builder()
                .tableName("customer")
                .partitionKey(Attribute.builder()
                        .name("customerKey")
                        .type(AttributeType.STRING)
                        .build())
                /*
                 * .sortKey(Attribute.builder()
                 * .name("pk")
                 * .type(AttributeType.STRING)
                 * .build())
                 */
                .build());
    }
}

File: nodejs/data/cdk-dynamo/lib/cdk-dynamo-stack.ts
import * as cdk from 'aws-cdk-lib';
import { AttributeType, Table } from 'aws-cdk-lib/aws-dynamodb';
import { Construct } from 'constructs';
// import * as sqs from 'aws-cdk-lib/aws-sqs';

export class CdkDynamoStack extends cdk.Stack {
  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new Table(this, "customer-table", {
      tableName: "customer",
      partitionKey: {
        name: "customerKey",
        type: AttributeType.STRING
      }
    });

  }
}
